data/mods/gen3colosseum/moves.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Gen 3 colosseum modifications (perish song should fail if it effects all remaining pokemon)
+ */
+export const Moves: import('../../../sim/dex-moves').ModdedMoveDataTable = {
+	perishsong: {
+		inherit: true,
+		onTryMove(attacker, defender, move) {
+			let soundproof = false;
+
+			for (const mon of attacker.side.active) {
+				if (mon.ability === "soundproof" || 'twoturnmove' in mon.volatiles) {
```

Comment:
Do all two-turn moves provide immunity? SolarBeam? Skull Bash?

Reply:
Fix live

Reply:
I think you could also just hardcode Dig and Fly and Dive. And maybe Bounce? Was that Gen 3?

Reply:
soundproof as well

Reply:
Overall, I prefer my current method as I don't always think of everything (like what happened with skull bash) and want to avoid any checklists if I can. Especially if we learn more about mechanics later like fly is not immune to perish song or something. The update to gen3 would cause this to work correctly without me having to update some extra list